doc/amoc_throttle.md

```diff
@@ -0,0 +1,152 @@
+## Overview
+
+Amoc throttle is a module that allows limiting number of users' actions per given interval, no matter how many users there are in a test.
+It works in both local and distributed environment, allows for dynamic rate changes during a test and exposes metrics which show number of requests and executions.
+
+Amoc throttle allows to set execution `Rate` per `Interval` or limit number of parallel executions when `Interval` is set to `0`.
+Each `Rate` is identified with a `Name`.
+Rate limiting mechanism is based on allowing execution in response to a request only when it would not exceed the given `Rate`.
```

Comment:
I think this would change the meaning of the sentence. What I meant was something like this:
"Rate limiting mechanism allows to execute a task in response to a request only when executing this task would not exceed the given `Rate`."
The described behaviour works like this:
1. `amoc_throttle` gets a request from a user. User requests a permission to execute an action.
2. `amoc_throttle` will always respond. The response may be sent after some time (waiting for when an execution would not exceed the `Rate` of those actions) or it may be instantaneous.

If you have any suggestions, I will be glad to hear them, because I'm not sure how to convey this mechanism clearly and it is the main idea behind `amoc_throttle`